src/QsCompiler/LlvmBindings/LLVMSharpExtensions/LLVMDIBuilderRefExtensions.cs

```diff
@@ -0,0 +1,198 @@
+        public static LLVMMetadataRef CreateNameSpace(this LLVMDIBuilderRef self, LLVMMetadataRef parentScope, string name, bool exportSymbols)
+        {
+            return LLVM.DIBuilderCreateNameSpace(self, parentScope, name.AsMarshaledString(), (UIntPtr)name.Length, exportSymbols ? 1 : 0);
+        }
+
+        /// <summary>Convenience wrapper for <see cref="LLVM.DIBuilderCreateLexicalBlock"/>.</summary>
+        public static LLVMMetadataRef CreateLexicalBlock(this LLVMDIBuilderRef self, LLVMMetadataRef scope, LLVMMetadataRef file, uint line, uint column)
+        {
+            return LLVM.DIBuilderCreateLexicalBlock(self, scope, file, line, column);
+        }
+
+        /// <summary>Convenience wrapper for <see cref="LLVM.DIBuilderCreateLexicalBlockFile"/>.</summary>
+        public static LLVMMetadataRef CreateLexicalBlockFile(this LLVMDIBuilderRef self, LLVMMetadataRef scope, LLVMMetadataRef file, uint discriminator)
+        {
+            return LLVM.DIBuilderCreateLexicalBlockFile(self, scope, file, discriminator);
+        }
+
+        /// <summary>Convenience wrapper for <see cref="LLVM.DIBuilderCreateAutoVariable"/>.</summary>
+        public static LLVMMetadataRef CreateAutoVariable(this LLVMDIBuilderRef self, LLVMMetadataRef scope, string name, LLVMMetadataRef file, uint line, LLVMMetadataRef ty, bool alwaysPreserve, LLVMDIFlags diflags, uint alignInBits)
+        {
+            return LLVM.DIBuilderCreateAutoVariable(self, scope, name.AsMarshaledString(), (UIntPtr)name.Length, file, line, ty, alwaysPreserve ? 1 : 0, diflags, alignInBits);
+        }
+
+        /// <summary>Convenience wrapper for <see cref="LLVM.DIBuilderCreateParameterVariable"/>.</summary>
+        public static LLVMMetadataRef CreateParameterVariable(this LLVMDIBuilderRef self, LLVMMetadataRef scope, string name, uint argNo, LLVMMetadataRef file, uint line, LLVMMetadataRef ty, bool alwaysPreserve, LLVMDIFlags diflags)
+        {
+            return LLVM.DIBuilderCreateParameterVariable(self, scope, name.AsMarshaledString(), (UIntPtr)name.Length, argNo, file, line, ty, alwaysPreserve ? 1 : 0, diflags);
+        }
+
+        /// <summary>Convenience wrapper for <see cref="LLVM.DIBuilderCreateBasicType"/>.</summary>
+        public static LLVMMetadataRef CreateBasicType(this LLVMDIBuilderRef self, string name, ulong sizeInBits, uint encoding, LLVMDIFlags diflags)
+        {
+            return LLVM.DIBuilderCreateBasicType(self, name.AsMarshaledString(), (UIntPtr)name.Length, sizeInBits, encoding, diflags);
+        }
+
+        /// <summary>Convenience wrapper for <see cref="LLVM.DIBuilderCreatePointerType"/>.</summary>
+        public static LLVMMetadataRef CreatePointerType(this LLVMDIBuilderRef self, LLVMMetadataRef pointeeTy, ulong sizeInBits, uint alignInBits, uint addressSpace, string? name)
+        {
+            if (name == null)
+            {
+                name = string.Empty;
```

Comment:
I haven't had my morning caffeine yet so may be interpreting this wrong, but wouldn't this have the side effect of changing the passed in string from null to empty string? If so, it would be better to use a local variable to avoid mutating the arguments. If not, please ignore!

Reply:
In this case, we should be good, since without the `ref` or `out` keywords, our parameter will be copied-in / passed by value.

Though, it may very well be against the style guidelines. If it is, I'll be sure to adjust!

Reply:
Makes sense, thanks!